samples/security/protection_psoc6/secure/src/main.c

```diff
@@ -0,0 +1,215 @@
+		status = ppu_prog_protect(prog_ppu_config,
+			ARRAY_SIZE(prog_ppu_config));
+	}
+	/* fixed group ppu */
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_gr_protect(fixed_gr_ppu_config,
+			ARRAY_SIZE(fixed_gr_ppu_config));
+	}
+
+	/* bus masters */
+	if (status == CY_PROT_SUCCESS) {
+		status = bus_masters_protect(bus_masters_config,
+			ARRAY_SIZE(bus_masters_config));
+	}
+
+	return status;
+}
+#else
+static cy_en_prot_status_t protect_cm0(void)
+{
+	cy_en_prot_status_t status = CY_PROT_SUCCESS;
+
+	/* smpu */
+	status = smpu_protect(flash_spm_smpu_config,
+		ARRAY_SIZE(flash_spm_smpu_config));
+	if (status == CY_PROT_SUCCESS) {
+		status = smpu_protect(sram_spm_smpu_config,
+			ARRAY_SIZE(sram_spm_smpu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = smpu_protect_unprotected(&spm_other_smpu_config);
+	}
+
+	/* fixed region ppu */
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_rg_protect(fixed_rg_pc0_ppu_config,
+			ARRAY_SIZE(fixed_rg_pc0_ppu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_rg_protect(fixed_rg_spm_ppu_config,
+			ARRAY_SIZE(fixed_rg_spm_ppu_config));
+	if (status == CY_PROT_SUCCESS) {
+	}
+		status = ppu_fixed_rg_protect(fixed_rg_any_ppu_config,
+			ARRAY_SIZE(fixed_rg_any_ppu_config));
+	}
+	/* fixed slave ppu */
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_sl_protect(fixed_sl_pc0_ppu_config,
+			ARRAY_SIZE(fixed_sl_pc0_ppu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_sl_protect(fixed_sl_spm_ppu_config,
+			ARRAY_SIZE(fixed_sl_spm_ppu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_sl_protect(fixed_sl_any_ppu_config,
+			ARRAY_SIZE(fixed_sl_any_ppu_config));
+	}
+	/* programmable ppu */
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_prog_protect(prog_pc0_ppu_config,
+			ARRAY_SIZE(prog_pc0_ppu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_prog_protect(prog_spm_ppu_config,
+			ARRAY_SIZE(prog_spm_ppu_config));
+	}
+	/* fixed group ppu */
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_gr_protect(fixed_gr_pc0_ppu_config,
+			ARRAY_SIZE(fixed_gr_pc0_ppu_config));
+	}
+	if (status == CY_PROT_SUCCESS) {
+		status = ppu_fixed_gr_protect(fixed_gr_spm_ppu_config,
+			ARRAY_SIZE(fixed_gr_spm_ppu_config));
+	}
+
+	/* bus masters */
+	if (status == CY_PROT_SUCCESS) {
+		status = bus_masters_protect(bus_masters_config,
+			ARRAY_SIZE(bus_masters_config));
+	}
+
+	return status;
+}
+#endif
+
+void main(void)
+{
+	u8_t access_flag;
+	u32_t *mem_addr;
+
+	/* Disable watchdog */
+	Cy_WDT_Unlock();
+	Cy_WDT_Disable();
+
+	printk("PSoC6 Secure core has started\n\n");
+
+	if (protect_cm0() == CY_PROT_SUCCESS) {
+		printk("PSoC6 CM0+ resources protected\n\n");
+	} else {
+		printk("PSoC6 CM0+ resourses protection failed\n");
+		while (1) {
+		}
+	}
+
+	printk("Check access rights before executing CM4 requests\n\n");
+	access_flag = isPeriferalAccessAllowed((u32_t)SCB2, REGION_SIZE,
```

Comment:
peripheral

Reply:
search and replace periferal w/ peripheral

Reply:
done